docker/src/test/java/org/jclouds/docker/compute/strategy/PredicateLocateImageByNameTest.java

```diff
@@ -0,0 +1,111 @@
+         "parent", // parent
+         new Date(), // created
+         "containerId", // container
+         "1.3.1", // dockerVersion
+         "x86_64", // architecture
+         "os", // os
+         0l, // size
+         0l, // virtualSize
+         ImmutableList.<String> of("kwart/alpine-ext:3.3-ssh", "kwart/alpine-ext:latest", "my-tag:latestdock") // repoTags
+   );
+
+   private static final Image IMAGE_REPO_TAGS_EMPTY = Image.create("id", // id
+         "author", "comment", Config.builder().image("imageId").build(), Config.builder().image("imageId").build(),
+         "parent", // parent
+         new Date(), // created
+         "containerId", // container
+         "1.3.1", // dockerVersion
+         "x86_64", // architecture
+         "os", // os
+         0l, // size
+         0l, // virtualSize
+         ImmutableList.<String> of() // repoTags
+   );
+
+   private static final Image IMAGE_REPO_TAGS_WITH_HOST = Image.create("id", // id
+         "author", "comment", Config.builder().image("imageId").build(), Config.builder().image("imageId").build(),
+         "parent", // parent
+         new Date(), // created
+         "containerId", // container
+         "1.3.1", // dockerVersion
+         "x86_64", // architecture
+         "os", // os
+         0l, // size
+         0l, // virtualSize
+         ImmutableList.<String> of("docker.io/kwart/alpine-ext:3.3-ssh", "docker.io/kwart/alpine-ext:latest") // repoTags
+   );
+
+   public void testRepoTagVersion() {
+      final Predicate<Image> predicate = DockerComputeServiceAdapter
+            .createPredicateMatchingRepoTags("kwart/alpine-ext:3.3-ssh");
+      assertTrue(predicate.apply(IMAGE_REPO_TAGS_MULTI));
+      assertFalse(predicate.apply(IMAGE_REPO_TAGS_EMPTY));
+      assertTrue(predicate.apply(IMAGE_REPO_TAGS_WITH_HOST));
+   }
+
+   public void testRepoTagLatest() {
+      final Predicate<Image> predicate = DockerComputeServiceAdapter.createPredicateMatchingRepoTags("kwart/alpine-ext");
+      assertTrue(predicate.apply(IMAGE_REPO_TAGS_MULTI));
+      assertFalse(predicate.apply(IMAGE_REPO_TAGS_EMPTY));
+      assertTrue(predicate.apply(IMAGE_REPO_TAGS_WITH_HOST));
+   }
+
+   public void testRepoTagVersionWithHost() {
+      final Predicate<Image> predicate = DockerComputeServiceAdapter
+            .createPredicateMatchingRepoTags("docker.io/kwart/alpine-ext:3.3-ssh");
+      assertFalse(predicate.apply(IMAGE_REPO_TAGS_MULTI));
+      assertFalse(predicate.apply(IMAGE_REPO_TAGS_EMPTY));
+      assertTrue(predicate.apply(IMAGE_REPO_TAGS_WITH_HOST));
+   }
+
+   public void testRepoTagLatestWithHost() {
+      final Predicate<Image> predicate = DockerComputeServiceAdapter
+            .createPredicateMatchingRepoTags("docker.io/kwart/alpine-ext");
+      assertFalse(predicate.apply(IMAGE_REPO_TAGS_MULTI));
+      assertFalse(predicate.apply(IMAGE_REPO_TAGS_EMPTY));
+      assertTrue(predicate.apply(IMAGE_REPO_TAGS_WITH_HOST));
+   }
+
+}
```

Comment:
Make sure your branch is properly rebased. This class shouldn't be part of the PR.